modules/internal/ChapelLocale.chpl

```diff
@@ -443,6 +443,16 @@ module ChapelLocale {
     @chpldoc.nodoc
     proc isGpu() : bool { return false; }
 
+    // if using a gpu locale return its position in the parent locale's
+    // `here.gpus` array
+    proc gpuId : int do return gpuIdImpl();
+
+    @chpldoc.nodoc
+    proc gpuIdImpl() : int {
+      halt("Can not use 'gpuId' field on a non gpu locale");
+      return -1;
+    }
+
```

Comment:
There's an unfortunate consequence of adding this here: they are not documented. I believe the locale documentation is in https://chapel-lang.org/docs/main/language/spec/locales.html#locale-methods, and it comes from the `record _locale` rather than this base implementation class, which is a forwarding field in the record. I believe Jade has recently mentioned that `isGpu` is also not documented. I believe the same applies for `proc gpus`.

This is not specific to this PR, and I think we can merge this as is, but could you open an issue about these methods that I think should be documented?

Reply:
Added here: https://github.com/Cray/chapel-private/issues/6758